pkg/login/ldap_settings.go

```diff
@@ -55,22 +56,28 @@ type LdapGroupToOrgRole struct {
 var LdapCfg LdapConfig
 var ldapLogger log.Logger = log.New("ldap")
 
-func loadLdapConfig() {
+func LoadLdapConfig(exitOnError bool) error {
 	if !setting.LdapEnabled {
-		return
+		return nil
 	}
 
 	ldapLogger.Info("Ldap enabled, reading config file", "file", setting.LdapConfigFile)
 
 	_, err := toml.DecodeFile(setting.LdapConfigFile, &LdapCfg)
 	if err != nil {
 		ldapLogger.Crit("Failed to load ldap config file", "error", err)
-		os.Exit(1)
+		if exitOnError {
+			os.Exit(1)
```

Comment:
Not for this PR, but in the future I think we should refactor this one layer up. Seems weird that something like `LoadLdapConfig` function is deciding whether to exit on error. cc @bergquist

Reply:
Agree. But let's solve that in another PR.